diff --git a/node_modules/@ory/nextjs/dist/app/index.js b/node_modules/@ory/nextjs/dist/app/index.js
index d4ade92..ed211cf 100644
--- a/node_modules/@ory/nextjs/dist/app/index.js
+++ b/node_modules/@ory/nextjs/dist/app/index.js
@@ -17,10 +17,10 @@ function getEnv(name) {
   return process.env[`NEXT_PUBLIC_${name}`] || process.env[name];
 }
 function orySdkUrl() {
-  const baseUrl = getEnv("ORY_SDK_URL");
+  const baseUrl = process.env["ORY_SDK_URL"];
   if (!baseUrl) {
     throw new Error(
-      "You need to set environment variable `NEXT_PUBLIC_ORY_SDK_URL` to your Ory Network SDK URL."
+      "You need to set environment variable `ORY_SDK_URL` to your Ory SDK URL."
     );
   }
   return baseUrl.replace(/\/$/, "");
@@ -42,6 +42,10 @@ function guessPotentiallyProxiedOrySdkUrl(options) {
       return process.env["__NEXT_PRIVATE_ORIGIN"].replace(/\/$/, "");
     }
   }
+  var publicUrl = process.env["ORY_SDK_PUBLIC_URL"];
+  if (publicUrl) {
+    return publicUrl.replace(/\/$/, "");
+  }
   if (isProduction()) {
     return orySdkUrl();
   }
diff --git a/node_modules/@ory/nextjs/dist/app/index.mjs b/node_modules/@ory/nextjs/dist/app/index.mjs
index da6b254..daf80f0 100644
--- a/node_modules/@ory/nextjs/dist/app/index.mjs
+++ b/node_modules/@ory/nextjs/dist/app/index.mjs
@@ -15,10 +15,10 @@ function getEnv(name) {
   return process.env[`NEXT_PUBLIC_${name}`] || process.env[name];
 }
 function orySdkUrl() {
-  const baseUrl = getEnv("ORY_SDK_URL");
+  const baseUrl = process.env["ORY_SDK_URL"];
   if (!baseUrl) {
     throw new Error(
-      "You need to set environment variable `NEXT_PUBLIC_ORY_SDK_URL` to your Ory Network SDK URL."
+      "You need to set environment variable `ORY_SDK_URL` to your Ory SDK URL."
     );
   }
   return baseUrl.replace(/\/$/, "");
@@ -40,6 +40,10 @@ function guessPotentiallyProxiedOrySdkUrl(options) {
       return process.env["__NEXT_PRIVATE_ORIGIN"].replace(/\/$/, "");
     }
   }
+  var publicUrl = process.env["ORY_SDK_PUBLIC_URL"];
+  if (publicUrl) {
+    return publicUrl.replace(/\/$/, "");
+  }
   if (isProduction()) {
     return orySdkUrl();
   }
